src/codecs/tga/decoder.rs

```diff
@@ -289,10 +290,13 @@ impl<R: Read + Seek> TgaDecoder<R> {
         let remain_len = self.line_remain_buff.len();
 
         if remain_len >= line_num_bytes {
-            let remain_buf = self.line_remain_buff.clone();
+            // `Vec::split_to` if std had it
+            let bytes = {
+                let bytes_after = self.line_remain_buff.split_off(line_num_bytes);
+                mem::replace(&mut self.line_remain_buff, bytes_after)
+            };
```

Comment:
I wonder hower `splice(..line_num_bytes, []).collect()` performs?

Reply:
I neven used `Vec::splice` and seeing how [bad the generated code is](https://rust.godbolt.org/z/YqfM973jE) I'll probably never use it :laughing:.
`Vec::split_off` is much much simpler, it just does [an allocation and a memcpy](https://doc.rust-lang.org/1.54.0/src/alloc/vec/mod.rs.html#1868-1901).